### PR TITLE
[861mu4f45][d3-chart] added cigarette chart

### PR DESCRIPTION
**Pull request by ilyabrower**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Nice improve.

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly or it's not required.
- [X] Unit tests are not broken.
- [X] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

**Comment by msereniti** [comment]
Screenshot 2024-04-10 at 16 00 42

Here little a11y module can really be inaccessible. Maybe we need to change it and give it some kind of minimal sizes?

File: website/docs/data-display/cigarette-chart/cigarette-chart-d3-code.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+## Basic usage
+
+::: sandbox
+
+<script lang="tsx">
+  export Demo from './examples/basic-usage.tsx';
+</script>
+
+:::
+
```
Comment (msereniti) [inline website/docs/data-display/cigarette-chart/cigarette-chart-d3-code.md:16]: Could also add an example of use without high level `Chart`?

File: website/docs/data-display/cigarette-chart/examples/basic-usage.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Chart } from '@semcore/d3-chart';
+
+class Demo extends React.PureComponent {
```
Comment (msereniti) [inline website/docs/data-display/cigarette-chart/examples/basic-usage.tsx:4]: Let's just use functional component here instead 😄

**Review by msereniti — APPROVED** [review]
Awesome work

But it would be very nice to also do the following:
1. Fix summary generating by removing the line.
2. Give a11y module minimal sizes to be displayed well with this chart.
3. Add an example of chart usage without the `Chart` component.
4. Remove useless pure components from this documentation examples.

File: website/docs/data-display/cigarette-chart/examples/basic-usage.tsx (hunk 2 above)
Comment (ilyabrower) [reply website/docs/data-display/cigarette-chart/examples/basic-usage.tsx:4]: Fixed. Copy-paste from some old example, I think

File: semcore/d3-chart/src/component/Cigarette/Cigarette.tsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      onMouseMove,
+      onMouseLeave,
+      transparent,
+      resolveColor,
+      patterns,
+    } = this.asProps;
+
+    const dSvg = this.getRect();
+
+    this.asProps.dataHintsHandler.describeValueEntity(dataKey, '');
```
[the remark this answers is not in the document]
Comment (ilyabrower) [reply semcore/d3-chart/src/component/Cigarette/Cigarette.tsx:117]: fixed, thank you

File: website/docs/data-display/cigarette-chart/cigarette-chart-d3-code.md (hunk 1 above)
Comment (ilyabrower) [reply website/docs/data-display/cigarette-chart/cigarette-chart-d3-code.md:16]: I'm not sure that we should. I didn't add export for CigaretteBar, right now we can use it only from high level Chart.
I've opened another PR (https://github.com/semrush/intergalactic/pull/1284) with example of advanced usage, but I'm not sure that we need to publishing it.

**Pull request merged**